lib/improver/tests/spotdata/spotdata/test_extract_data.py

```diff
@@ -285,25 +311,61 @@ def test_invalid_method(self):
             plugin.process(cube, self.sites, self.neighbour_list, {}, None,
                            **self.kwargs)
 
-    def test__build_coordinates(self):
-        """
-        Test the _build_coordinates private function.
 
+class Test__build_coordinate(Test_setup):
+    """Test the _build_coordinate function."""
+
+    def test_build_latitude_coordinate(self):
         """
-        plugin = Plugin()._build_coordinates
-        points = np.array([0, 1, 2])
-        indices, latitude, longitude, altitude, utc_offset, wmo_site = (
-            plugin(points, points, points, points[::-1], points))
-        self.assertArrayEqual(indices.points, points)
-        self.assertArrayEqual(altitude.points, points)
-        self.assertArrayEqual(utc_offset.points, points[::-1])
-        self.assertEqual(latitude.name(), 'latitude')
-        self.assertEqual(longitude.name(), 'longitude')
-        self.assertEqual(wmo_site.name(), 'wmo_site')
-        self.assertIsInstance(indices, DimCoord)
-        self.assertIsInstance(latitude, AuxCoord)
+        Test buiding a latitude coordinate.
 
-    def test_make_cube(self):
+        """
+        plugin = Plugin()._build_coordinate
+        coord_system = iris.coord_systems.GeogCS(6371229.0)
+        result = plugin(self.latitudes, 'latitude', units='degrees',
+                        coord_system=coord_system)
+        self.assertArrayEqual(result.points, self.latitudes)
+        self.assertEqual(result.name(), 'latitude')
+        self.assertIsInstance(result, DimCoord)
+        self.assertEqual(result.units, 'degrees')
+
+
+class Test_make_stat_coordinate_first(Test_setup):
+    """Test the function to reorder a cube to ensure the statistical coordinate
+    is the first dimension."""
+
+    def test_cube_reorder_realization(self):
+        """Test reordering a cube with a realization coordinate to make it come
+        first."""
+        plugin = Plugin().make_stat_coordinate_first
+        cube = set_up_cube(
+            zero_point_indices=(
+                (0, 0, 2, 2), (1, 0, 3, 3), (0, 1, 0, 0), (1, 1, 2, 1)),
+            num_time_points=2, num_grid_points=5, num_realization_points=2)
+        incorrect_cube = cube.copy()
+        incorrect_cube.transpose([1, 2, 0, 3])
+        result = plugin(incorrect_cube)
+        self.assertEqual(cube.coords()[0].name(), result.coords()[0].name())
```

Comment:
I've added, but left the original to emphasise that it is the first coordinate in which we are most interested.